src/Entity/EntityIdValue.php

```diff
@@ -31,8 +31,8 @@ public function __construct( EntityId $entityId ) {
 	 */
 	public function serialize() {
 		return json_encode( [
-			$this->entityId->getEntityType(),
-			$this->getNumericId()
+			get_class( $this->entityId ),
```

Comment:
For what is worth: if we didn't want to use PHP serialization for EntityIdValue that @thiemowmde is suggesting in https://github.com/wmde/WikibaseDataModel/pull/718 (I am not sure what are implications with that approach, if this serialization is never stored) another option (theoretically) could be using `EntityIdComposer` which is meant as a entity-type-agnostic (kinda) replacement for `LegacyIdInterpreter`. The obvious problem is how to pull `EntityIdComposer` in here, so it makes sense (it is now bound to Wikibase git repo and entity type definitions out there). I cannot think of good way to achieve this, so this comment is mostly noise.

Also theoretically, another option would be to use value of EntityId::getSerialization in EntityIdValue::serialize, and use EntityIdParser in EntityIdValue::unserialize but that seems so wrong.